htdocs/product/composition/card.php

```diff
@@ -407,17 +423,20 @@
 						print '<td>'.($value['incdec'] == 1 ? 'x' : '').'</td>';
 					}
 
+					print '<td class="linecolmove tdlineupdown center">';
+					print '</td>';
 					print '</tr>'."\n";
 				} else {
 					$hide = '';
 					if (empty($conf->global->PRODUCT_SHOW_SUB_SUB_PRODUCTS)) {
 						$hide = ' hideobject'; // By default, we do not show this. It makes screen very difficult to understand
 					}
 
-					print '<tr class="oddeven'.$hide.'" id="sub-'.$value['id_parent'].'">';
+					print '<tr class="oddeven'.$hide.'" id="sub" parent_id="'.$value['id_parent'].'">';
```

Comment:
The value for id="sub" means you will hame several time the same id on several lines. This is not allowed.
Why removing 
id="sub-'.$value['id_parent'].'">';
?

Reply:
When the configuration PRODUCT_SHOW_SUB_SUB_PRODUCTS is not activated : 
![image](https://user-images.githubusercontent.com/67913809/131808055-1e3a404b-195b-4d39-a4d2-d8e2ef720f2c.png)

When  the configuration PRODUCT_SHOW_SUB_SUB_PRODUCTS is activated:
![image](https://user-images.githubusercontent.com/67913809/131808369-962e6d8a-a886-4417-af94-ba92d0239ff3.png)

Regardless of the configuration, the sub-lines are still created, with the "hidden" state or not.
As you can see, if we leave the code as it is (keeping the parent id for the sub-lines), then these lines will be taken into account when calculating the ranks and this is a problem

Reply:
May be the code is working. But you introduce a bug in html validation.
There is several time the same id into the html page. Where id can have onl one value.

Reply:
Is it really usefull to have an id for these lines when they have an attribute parent-id ? 
Can I remove it ? that will avoid having several times id = "sub" on different lines and that will not impact the functioning of the development

Reply:
Why removing it ? Why not keeping it with a unique name ?

Reply:
Because when we have an integer id for these sub-line, these will be taken for the rank count.
As you can see, on pictures above, sub-lines take ranks and disorder all the list.